sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,1739 @@
+        //        }
+        //
+        //        // Restore the state of the reader
+        //        c.pop();
+        //
+        //        // Return our record
+        //        Ok(C2PA {
+        //            majorVersion: internal_record.majorVersion,
+        //            minorVersion: internal_record.minorVersion,
+        //            active_manifest_uri: active_manifest_uri,
+        //            manifestStore: manifest_store,
+        //        })
+        //    }
+        //}
+    }
+
+    /// Get the manifest store data if available
+    pub fn get_manifest_store(&self) -> Option<&[u8]> {
+        self.manifest_store.as_deref()
+    }
+
+    /// Serialize this C2PA table to the given writer.
+    fn write<TDest: Write + ?Sized>(&mut self, destination: &mut TDest) -> Result<()> {
+        // Set up the structured data
+        let mut raw_table = TableC2PARaw {
+            majorVersion: self.major_version,
+            minorVersion: self.minor_version,
+            activeManifestUriOffset: 0,
+            activeManifestUriLength: 0,
+            manifestStoreOffset: 0,
+            manifestStoreLength: 0,
+        };
+        // If a remote URI is present, prepare to store it.
+        if let Some(uri_string) = self.active_manifest_uri.as_ref() {
+            raw_table.activeManifestUriOffset = size_of::<TableC2PARaw>() as u32;
+            raw_table.activeManifestUriLength = uri_string.len() as u16;
+        }
+        // If a local store is present, prepare to store it.
+        if let Some(manifest_store) = self.manifest_store.as_ref() {
+            raw_table.manifestStoreOffset = size_of::<TableC2PARaw>() as u32 + raw_table.activeManifestUriOffset + raw_table.activeManifestUriLength as u32;
+            raw_table.manifestStoreLength = manifest_store.len() as u32;
+        }
+        // Write the table data
+        raw_table.write(destination)?;
+        // Write the remote manifest URI, if present.
+        if let Some(uri_string) = self.active_manifest_uri.as_ref() {
+            destination.write_all(uri_string.as_bytes())?;
+        }
+        // Write out the local manifest store, if present.
+        if let Some(manifest_store) = self.manifest_store.as_ref() {
+            destination.write_all(manifest_store)?;
+        }
+        // Done
+        Ok(())
+    }
+}
+
+impl Default for TableC2PA {
+    fn default() -> Self {
+        Self {
+            major_version: 0,
+            minor_version: 1,
+            active_manifest_uri: Default::default(),
+            manifest_store: Default::default(),
+        }
+    }
+}
+
+/// 'head' font table. For now, there is no need for a 'raw' variant, since only
+/// byte-swapping is needed.
+#[derive(Debug)]
+#[repr(C, packed)]       // As defined by Open Font Format / OpenType (though we don't as yet directly support exotics like FIXED).
+#[allow(non_snake_case)] // As named by Open Font Format / OpenType.
+struct TableHead {
```

Comment:
Well, `head` tables will usually be present, and the WOFF spec indicates that checksums should be correct (from ["Data Tables"](https://www.w3.org/TR/WOFF/#DataTables) in the WOFF1 spec:

> A WOFF file contains the same set of font tables as the input font from which it was created. This means that the overall font checksum of a font decompressed from a conformant WOFF file will always match the checksum in the well-formed input font. In the case where the input font included unreferenced data between or after the actual tables, this would affect the overall checksum of the input font, but would be dropped during creation of the WOFF file.

> A well-formed input font does not have structural anomalies such as incorrect padding, overlapping font tables, extraneous data between tables (which will be discarded by the WOFF generator), or incorrect checksums.

> **Checksums are correct**
> The WOFF specification says that table checksums must be validated (and corrected if necessary) by WOFF creators. In order for complete round-trip fidelity, therefore, the checksums in the input sfnt file must also be correct prior to WOFF packaging.